lib/compilers/hlsl.js

```diff
@@ -35,17 +42,79 @@ export class HLSLCompiler extends BaseCompiler {
         super(info, env);
 
         this.compiler.supportsIntel = false;
+
+        if (default_dxc === null && this.compiler.group === 'dxc') {
+            default_dxc = this.compiler;
+        }
+
+        if (this.rga) {
+            // RGA is invoked in two steps. First, DXC is invoked to compile
+            // the SPIR-V output of the HLSL file. Next, RGA is invoked to
+            // consume the SPIR-V output and produce the requested ISA. To
+            // accomplish this two-pass compilation, a helper script rga.js
+            // is used to wrap the compilation as a single executable script.
+            this.rgaOptions = [
+                path.join(path.dirname(url.fileURLToPath(import.meta.url)), 'rga.js'),
+                this.compiler.exe,
+            ];
+
+            // Replace the RGA executable with the node path
+            this.compiler.exe = process.execPath;
+        }
+    }
+
+    // Returns true if the compiler selected is a Radeon GPU Analyzer
+    get rga() {
+        return this.compiler.group === 'rga';
```

Comment:
As above; please use another way. We can chat to find a good solution. To me it seems like `rga` is a different type of compiler all together (that is, this is what `compilerType` which selects the appropriate `.js` file is for).

Reply:
definitely should be a different compilerType, that would be most consistent with the rest of CE

Reply:
Sounds good will grep the codebase for some examples, thanks!